<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Evan Teran</author>
 <class>DialogOptions</class>
 <widget class="QDialog" name="DialogOptions">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>602</width>
    <height>387</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Configure Debugger</string>
  </property>
  <layout class="QVBoxLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout">
       <property name="spacing">
        <number>6</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Close Behaviour</string>
         </property>
         <layout class="QVBoxLayout">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <widget class="QRadioButton" name="rdoDetach">
            <property name="text">
             <string>Detach From Debugged Application</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="rdoKill">
            <property name="text">
             <string>Terminate Debugged Application</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Appearance</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0" colspan="2">
        <layout class="QGridLayout">
         <item row="0" column="1">
          <widget class="QLineEdit" name="txtStackFont"/>
         </item>
         <item row="3" column="2">
          <widget class="QToolButton" name="btnDisFont">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="txtRegisterFont"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="lblDisFont">
           <property name="text">
            <string>Default Disassembly View Font</string>
           </property>
           <property name="buddy">
            <cstring>txtDisFont</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="lblStackFont">
           <property name="text">
            <string>Default Stack View Font</string>
           </property>
           <property name="buddy">
            <cstring>txtStackFont</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="lblRegisterFont">
           <property name="text">
            <string>Default Register View Font</string>
           </property>
           <property name="buddy">
            <cstring>txtRegisterFont</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="txtMemoryFont"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="lblMemFont">
           <property name="text">
            <string>Default Memory Dump View Font</string>
           </property>
           <property name="buddy">
            <cstring>txtMemoryFont</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QToolButton" name="btnRegisterFont">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QToolButton" name="btnStackFont">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLineEdit" name="txtDisFont"/>
         </item>
         <item row="1" column="2">
          <widget class="QToolButton" name="btnMemoryFont">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>Default Data View Settings</string>
         </property>
         <layout class="QGridLayout">
          <item row="0" column="0" colspan="3">
           <widget class="QCheckBox" name="chkDataShowAddress">
            <property name="text">
             <string>Show Address</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="3">
           <widget class="QCheckBox" name="chkDataShowHex">
            <property name="text">
             <string>Show Hex</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="3">
           <widget class="QCheckBox" name="chkDataShowAscii">
            <property name="text">
             <string>Show ASCII</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="3">
           <widget class="QCheckBox" name="chkDataShowComments">
            <property name="text">
             <string>Show Comments</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Word Width</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QComboBox" name="cmbDataWordWidth">
            <item>
             <property name="text">
              <string>1 Byte</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2 Bytes</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>4 Bytes</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>8 Bytes</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="4" column="2">
           <spacer>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>321</width>
              <height>25</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Row Width</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QComboBox" name="cmbDataRowWidth">
            <item>
             <property name="text">
              <string>1 Byte</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2 Bytes</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>4 Bytes</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>8 Bytes</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>16 Bytes</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="5" column="2">
           <spacer>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>321</width>
              <height>25</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QCheckBox" name="chkAddressSemicolon">
           <property name="text">
            <string>Show Semicolon in Addresses</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="2" column="0" colspan="2">
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>578</width>
           <height>201</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Debugging</string>
      </attribute>
      <layout class="QVBoxLayout">
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Initial Breakpoint</string>
         </property>
         <layout class="QVBoxLayout">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <widget class="QRadioButton" name="rdoBPEntry">
            <property name="text">
             <string>Application Entry Point</string>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="rdoBPMain">
            <property name="text">
             <string>&quot;main&quot; symbol (when possible)</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chkWarnDataBreakpoint">
         <property name="text">
          <string>Warn when setting breakpoint in non-executable region</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chkFindMain">
         <property name="text">
          <string>Use heuristic to find &quot;main&quot; symbol when it is not found in symbol file</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Minimum length for string detection</string>
           </property>
           <property name="buddy">
            <cstring>spnMinString</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spnMinString"/>
         </item>
         <item>
          <spacer>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Command Line IO</string>
         </property>
         <layout class="QGridLayout">
          <item row="0" column="0" colspan="3">
           <widget class="QCheckBox" name="chkTTY">
            <property name="text">
             <string>Open a terminal to provide program input and output</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Terminal Program:</string>
            </property>
            <property name="buddy">
             <cstring>txtTTY</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="txtTTY">
            <property name="text">
             <string>/usr/bin/xterm</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QToolButton" name="btnTTY">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>578</width>
           <height>31</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_6">
      <attribute name="title">
       <string>Signals/Exceptions</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Ignore (pass to program) the following exceptions:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="listIgnoredExceptions">
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Disassembly</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Disassembly Syntax</string>
         </property>
         <layout class="QVBoxLayout">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <widget class="QRadioButton" name="rdoSytntaxIntel">
            <property name="text">
             <string>Intel</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="rdoSytntaxATT">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>AT&amp;&amp;T</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chkZerosAreFilling">
         <property name="text">
          <string>Instruction &quot;add byte ptr[eax], al&quot; (0x00 0x00) is &quot;Filling&quot; on x86</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chkUppercase">
         <property name="text">
          <string>Disassemble in uppercase</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>561</width>
           <height>206</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_5">
      <attribute name="title">
       <string>Directories</string>
      </attribute>
      <layout class="QVBoxLayout">
       <item>
        <layout class="QGridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Symbol Directory</string>
           </property>
           <property name="buddy">
            <cstring>txtSymbolDir</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="txtSymbolDir"/>
         </item>
         <item row="0" column="2">
          <widget class="QToolButton" name="btnSymbolDir">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Plugin Directory</string>
           </property>
           <property name="buddy">
            <cstring>txtPluginDir</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="txtPluginDir"/>
         </item>
         <item row="1" column="2">
          <widget class="QToolButton" name="btnPluginDir">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Session Directory</string>
           </property>
           <property name="buddy">
            <cstring>txtSessionDir</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLineEdit" name="txtSessionDir"/>
         </item>
         <item row="2" column="2">
          <widget class="QToolButton" name="btnSessionDir">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>578</width>
           <height>101</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_plugins">
      <attribute name="title">
       <string>Plugin Options</string>
      </attribute>
      <layout class="QGridLayout" name="tab_plugins_layout">
       <item row="0" column="0">
        <widget class="QLabel" name="label_plugins">
         <property name="text">
          <string>No Plugin Option Pages</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>rdoDetach</tabstop>
  <tabstop>rdoKill</tabstop>
  <tabstop>txtStackFont</tabstop>
  <tabstop>btnStackFont</tabstop>
  <tabstop>txtMemoryFont</tabstop>
  <tabstop>btnMemoryFont</tabstop>
  <tabstop>txtRegisterFont</tabstop>
  <tabstop>btnRegisterFont</tabstop>
  <tabstop>txtDisFont</tabstop>
  <tabstop>btnDisFont</tabstop>
  <tabstop>chkDataShowAddress</tabstop>
  <tabstop>chkDataShowHex</tabstop>
  <tabstop>chkDataShowAscii</tabstop>
  <tabstop>chkDataShowComments</tabstop>
  <tabstop>cmbDataWordWidth</tabstop>
  <tabstop>cmbDataRowWidth</tabstop>
  <tabstop>rdoBPEntry</tabstop>
  <tabstop>rdoBPMain</tabstop>
  <tabstop>chkWarnDataBreakpoint</tabstop>
  <tabstop>chkFindMain</tabstop>
  <tabstop>spnMinString</tabstop>
  <tabstop>chkTTY</tabstop>
  <tabstop>txtTTY</tabstop>
  <tabstop>btnTTY</tabstop>
  <tabstop>rdoSytntaxIntel</tabstop>
  <tabstop>rdoSytntaxATT</tabstop>
  <tabstop>chkZerosAreFilling</tabstop>
  <tabstop>chkUppercase</tabstop>
  <tabstop>txtSymbolDir</tabstop>
  <tabstop>btnSymbolDir</tabstop>
  <tabstop>txtPluginDir</tabstop>
  <tabstop>btnPluginDir</tabstop>
  <tabstop>txtSessionDir</tabstop>
  <tabstop>btnSessionDir</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>chkTTY</sender>
   <signal>toggled(bool)</signal>
   <receiver>txtTTY</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogOptions</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>472</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogOptions</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>559</x>
     <y>411</y>
    </hint>
    <hint type="destinationlabel">
     <x>539</x>
     <y>397</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkTTY</sender>
   <signal>toggled(bool)</signal>
   <receiver>btnTTY</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>241</y>
    </hint>
    <hint type="destinationlabel">
     <x>567</x>
     <y>267</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
